src/store-api/src/storage/region.rs

```diff
@@ -110,10 +110,40 @@ pub struct FlushContext {
     /// If true, the flush will wait until the flush is done.
     /// Default: true
     pub wait: bool,
+    /// Flush reason.
+    pub reason: FlushReason,
 }
 
 impl Default for FlushContext {
     fn default() -> FlushContext {
-        FlushContext { wait: true }
+        FlushContext {
+            wait: true,
+            reason: FlushReason::Unknown,
+        }
+    }
+}
+
+/// Reason of flush operation.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FlushReason {
+    /// Unknown reason.
+    Unknown,
```

Comment:
When is a flush reason "unknown"?

Reply:
Normally there won't be an unknown reason. This is a placeholder for a default flush reason in FlushContext. Maybe I can remove it as the `FlushContext::default` is never used.

Reply:
The default is used in tests, I still plan to keep this reason but rename it to `others`.